src/ralph/api/routers/statements.py

```diff
@@ -416,19 +423,21 @@ async def put(
     _enrich_statement_with_authority(statement_as_dict, current_user)
 
     try:
-        existing_statement = DATABASE_CLIENT.query_statements_by_ids([statementId])
+        existing_statements = desync(
```

Comment:
why is `desync` necessary (if it only applies to synchronous generators) ?

Reply:
It is a wrap to be able to iterator over either a synchronous or asynchronous generator ([https://stackoverflow.com/a/67931234/10512887](https://stackoverflow.com/a/67931234/10512887))